.github/workflows/tests.yml

```diff
@@ -0,0 +1,46 @@
+name: tests
+
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '.gitignore'
+      - '**.md'
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - '.gitignore'
+      - '**.md'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python_version: [
+            '3.7',
+#            3.8,  # only lowest & highest version should be enough, :cross_fingers:
+#            3.9,
+#            '3.10',
+            '3.11'
+        ]
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Python ${{ matrix.python_version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python_version }}
+
+    - name: Install dependencies
+      run: pip install -r requirements.txt
+
+    - name: style & lint checks
+      run: pre-commit run --all-files
+
+    - name: pytest
+      run: pytest -v
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
whenever you see these red circles in the diff, it means it's missing  a newline at the end. it's good practice to add that